From https://github.com/tamasfe/taplo/pull/354
From: Michael Davis <mcarsondavis@gmail.com>
Date: Sun, 20 Nov 2022 12:25:43 -0600
Subject: [PATCH] Handle LSP shutdown request

These lines in the Server loop discard any messages with the shutdown
method. The Server can handle and correctly respond to shutdown
requests, though, so there's no need to discard the message.
---
 crates/lsp-async-stub/src/listen.rs | 2 --
 1 file changed, 2 deletions(-)

diff --git a/crates/lsp-async-stub/src/listen.rs b/crates/lsp-async-stub/src/listen.rs
index fca0a2980..b6b00f5fd 100644
--- a/crates/lsp-async-stub/src/listen.rs
+++ b/crates/lsp-async-stub/src/listen.rs
@@ -64,8 +64,6 @@ impl<W: Clone + 'static> Server<W> {
                                     drop(output);
                                     drop(input);
                                     break;
-                                } else if msg.method.as_ref().map(|m| m == "shutdown").unwrap_or(false) {
-                                    continue;
                                 }
 
                                 let task_fut = self.handle_message(

